test/rng_testsuite/extra_methods_tests/engines_methods.pass.cpp

```diff
@@ -0,0 +1,263 @@
+
+        try
+        {
+            queue.submit([&](sycl::handler& cgh) {
+                auto dpstd_acc = dpstd_buffer.template get_access<sycl::access::mode::write>(cgh);
+
+                cgh.parallel_for<>(sycl::range<1>(N_GEN), [=](sycl::item<1> idx) {
+                    unsigned long long offset = idx.get_linear_id() * num_elems;
+                    Engine engine0(SEED);
+                    Engine engine1;
+                    engine1.seed(SEED);
+                    engine0.discard(offset);
+                    engine1.discard(offset);
+                    typename Engine::result_type res0;
+                    if constexpr (std::is_same<Engine, oneapi::dpl::ranlux24>::value)
+                    {
+                        Engine engine(engine1);
+                        auto eng = engine.base();
+                        res0 = engine();
+                    }
+                    else
+                    {
+                        res0 = engine0();
+                    }
+                    typename Engine::result_type res1 = engine1();
+                    if constexpr ((num_elems > 1) || (std::is_same<typename Engine::result_type,
+                                                                   sycl::vec<typename Engine::scalar_type, 1>>::value))
+                    {
+                        std::int32_t is_inequal = 0;
+                        for (std::int32_t i = 0; i < num_elems; ++i)
+                        {
+                            if (res0[i] != res1[i])
+                            {
+                                is_inequal = 1;
+                            }
+                        }
+                        dpstd_acc[offset] = is_inequal;
+                    }
+                    else
+                    {
+                        if (res0 != res1)
+                        {
+                            dpstd_acc[offset] = 1;
+                        }
+                        else
+                        {
+                            dpstd_acc[offset] = 0;
+                        }
+                    }
+                });
+            });
+            auto dpstd_acc = dpstd_buffer.template get_access<sycl::access::mode::write>();
+            for (int i = 0; i < N_GEN; ++i)
+            {
+                sum += dpstd_acc[i];
+            }
+            if (sum)
+            {
+                std::cout << "Error occured in " << sum << " elements" << std::endl;
+            }
+        }
+        catch (sycl::exception const& e)
+        {
+            std::cout << "\t\tSYCL exception during generation\n"
+                      << e.what() << std::endl
+                      << "OpenCL status: " << e.get_cl_code() << std::endl;
+            return 1;
+        }
+
+        queue.wait_and_throw();
+        Engine engine;
+        sum += check_params<Engine, num_elems>(engine);
+    }
+
+    return sum;
+}
+
+#endif // _ONEDPL_BACKEND_SYCL
+
+int
+main()
+{
+
+#if _ONEDPL_BACKEND_SYCL

```

Comment:
Please look at the freshest main branch and changes that are done by https://github.com/oneapi-src/oneDPL/pull/182
Here should be no usage of `_ONEDPL_BACKEND_SYCL`

Reply:
changed to TEST_DPCPP_BACKEND_PRESENT